src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.h

```diff
@@ -0,0 +1,209 @@
+/**
+ * The Visualizer messenger is a singleton object that receives draw calls
+ * to draw shapes such as ellipse and rectangle.
+ *
+ * The messenger constructs and queues the shapes into shape messages and layer
+ * messages to be sent via ROS visualizer topics (VISUALIZER_DRAW_LAYER_TOPIC)
+ *
+ * Lasted edited by Muchen He on 2019-01-28
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "thunderbots_msgs/DrawLayer.h"
+#include "thunderbots_msgs/DrawShape.h"
+
+namespace Util
+{
+    using LayerMsg    = thunderbots_msgs::DrawLayer;
+    using LayerMsgMap = std::map<std::string, LayerMsg>;
+    using ShapeMsg    = thunderbots_msgs::DrawShape;
+
+    class VisualizerMessenger
+    {
+       public:
+        /**
+         * DrawStyle is a struct that packs the appearance properties that goes into a
+         * shape message The default constructor contains the default value of the draw
+         * style
+         */
+        typedef struct DrawStyle
+        {
+            // Struct constructor
+            DrawStyle() : fill("white"), stroke("black"), stroke_weight(1) {}
+
+            std::string fill;
+            std::string stroke;
+            uint8_t stroke_weight;
+        } DrawStyle;
+
+        /**
+         * DrawTransform is a struct that packs the transform properties that goes into a
+         * shape message The default constructor contains the default value of the
+         * transformation
+         */
+        typedef struct DrawTransform
+        {
+            DrawTransform() : rotation(0.0), scale(1.0) {}
+
+            double rotation;
+            double scale;
+        } DrawTransform;
+
+       public:
+        /**
+         * Getter of the singleton object.
+         *
+         * @return A shared pointer of the static instance
+         */
+        static std::shared_ptr<VisualizerMessenger> getInstance()
```

Comment:
static functions should still be implemented in the `.cpp` file for consistency.